Fix talloc linking.

It seems impossible to mix static libraries with dynamic libraries, at
least under the “foo.a bar.a -lbaz” form. Also, no combination of
pkg-config/--libs/--static/talloc seems to return the wanted
“/usr/lib/libtalloc.a”, one only gets “-ltalloc”.

So hardcode the path for now, using another variable. This way,
$(TALLOC_LIBS) might be added to other places where dynamic variables
are OK, and we don't break that.

Signed-off-by: Cyril Brulebois <kibi@debian.org>

--- a/configs/default
+++ b/configs/default
@@ -84,6 +84,7 @@ VG_LIB_GLOB = $(VG_LIB_NAME)*
 
 TALLOC_LIBS = `pkg-config --libs talloc`
 TALLOC_CFLAGS = `pkg-config --cflags talloc`
+TALLOC_STATIC_LIBS = /usr/lib/libtalloc.a
 
 # Optional assembly language optimization files for libGL
 MESA_ASM_SOURCES = 
--- a/src/mesa/drivers/osmesa/Makefile
+++ b/src/mesa/drivers/osmesa/Makefile
@@ -40,7 +40,7 @@ $(TOP)/$(LIB_DIR)/$(OSMESA_LIB_NAME): $(
 		-major 6 -minor 5 -patch 3 \
 		-install $(TOP)/$(LIB_DIR) -cplusplus $(MKLIB_OPTIONS) \
 		-id $(INSTALL_LIB_DIR)/lib$(OSMESA_LIB).$(MESA_MAJOR).dylib \
-		$(OSMESA_LIB_DEPS) $(OBJECTS) $(CORE_MESA) $(TALLOC_LIBS)
+		$(OSMESA_LIB_DEPS) $(OBJECTS) $(CORE_MESA) $(TALLOC_STATIC_LIBS)
 
 
 
--- a/src/glsl/Makefile
+++ b/src/glsl/Makefile
@@ -84,7 +84,7 @@ CXX_SOURCES = \
 
 LIBS = \
 	$(TOP)/src/glsl/libglsl.a \
-	$(TALLOC_LIBS)
+	$(TALLOC_STATIC_LIBS)
 
 APPS = glsl_compiler glcpp/glcpp
 
